build.py

```diff
@@ -0,0 +1,179 @@
+#! /bin/python3
+import argparse
+import contextlib
+import itertools
+import json
+import os
+import shutil
+import subprocess
+import sys
+import time
+from functools import partial
+from glob import glob
+from pathlib import Path
+from zipfile import ZipFile, ZIP_DEFLATED
+
+printe = partial(print, file=sys.stderr)
+
+try:
+    import watchdog
+except ModuleNotFoundError:
+    watchdog = None  # Not mandatory
+
+
+EXTENSION_ZIP_NAME = 'NopeCHA.zip'
+BASE_MANIFEST = Path('manifest.base.json')
+EXPORT_PATH = Path('__export__')
+VERSIONS_PATH = Path('version')
+
+
+watchdog_help = (f"""
+Keeps the program running and makes use of the watchdog library to listen for changes.
+When changes exist, the corresponding file is updated in {EXPORT_PATH} for the extension to update itself and use.
+""").strip()
+
+WATCHDOG_NOT_FOUND = f"""
+    watchdog was not found. Execute the command below if you want to use the -w/--watch feature:
+    python -m pip install -t pylibs watchdog
+""".strip()
+
+if watchdog is None:
+    watchdog_help += f"""
+Warning:
+{WATCHDOG_NOT_FOUND}
+"""
+
+parser = argparse.ArgumentParser(
+    description='Dev/deploy build extension for firefox and chrome',
+    formatter_class=argparse.RawTextHelpFormatter
+)
+
+parser.add_argument('-p', '--production', action="store_true", help="Compresses files and creates zip file to submit for evaluation or test in the browser")
+parser.add_argument('-w', '--watch', action="store_true", help=watchdog_help)
+
+program_args = parser.parse_args()
+
+if watchdog is None and program_args.watch:
+    parser.print_help()
+    printe(f"\nERROR: \n {WATCHDOG_NOT_FOUND}\n")
+    exit(2)
+
+@contextlib.contextmanager
+def in_dir(new_path):
+    oldcwd = os.getcwd()
+    try:
+        os.chdir(new_path)
+        yield
+    finally:
+        os.chdir(oldcwd)
+
+dir_path = Path(__file__).absolute().parent
+
+with in_dir(dir_path):
+
+    def deploy():
+        here_path = Path('.')
+
+        # This is slow but there should be very few files, so it should be OK
+        files_to_include = list(itertools.chain(
+            here_path.glob('*.html'),
+            here_path.glob('*.css'),
+            here_path.glob('*.js'),
+            (here_path / 'icon').iterdir(),
+        ))
+
+        versions = filter(lambda p: p.is_dir(), VERSIONS_PATH.iterdir())
+        # This could be run just once per program run but it would bring unnecessary complexity to the code below
+        shutil.rmtree(EXPORT_PATH, True)
+
+        for version in versions:
+            export_directory = EXPORT_PATH / version
+            export_directory.mkdir(parents=True, exist_ok=True)
+
+            if program_args.production:
+                # Takes quite a while to make new zips. So better do it for final tests or when deploying
+                zip_deploy = ZipFile(export_directory / EXTENSION_ZIP_NAME, 'w', ZIP_DEFLATED, compresslevel=9)
```

Comment:
```suggestion
                ext = "xpi" if version.basename == "firefox" else "crc"
                zip_deploy = ZipFile(EXPORT_PATH / f"NopeCHA.{ext}", 'w', ZIP_DEFLATED, compresslevel=9)
```
put them in the main `dist` directory with the correct file extension

Reply:
I usually use conversion dictionaries for this, even when there's just 2 expected values. However, I think this is fine.

Reply:
feel free to 🤷